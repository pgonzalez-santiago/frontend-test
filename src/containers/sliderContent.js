import React from 'react'
import PropTypes from 'prop-types'
import { NavLink } from 'react-router-dom'

// Components
import GithubLogo from '../components/githubLogo'

// Styles
import './styles/sliderContent.scss'
import colors from '../assets/colors'

import styled from 'styled-components'

const Logo = styled.div`
  padding-top: 10px;
  padding-bottom: 20px;
  border-bottom: 1px solid ${colors.primaryColor};
`

// Slider content example
const SliderContent = ({ onItemClick }) => {
  return (
    <div className="slider-content bm-item-list">
      <Logo>
        <GithubLogo
          width={40}
          height={40}
          style={{
            color:   colors.primaryColor,
            display: 'block',
            fill:    'currentColor',
            margin:  '0 auto',
          }}/>
      </Logo>
      <NavLink
        activeStyle={{ fontWeight: 'bold' }}
        onClick={onItemClick}
        to="/react">
        {'react'}
      </NavLink>
      <NavLink
        activeStyle={{ fontWeight: 'bold' }}
        onClick={onItemClick}
        to="/react-native">
        {'react-native'}
      </NavLink>
      <NavLink
        activeStyle={{ fontWeight: 'bold' }}
        onClick={onItemClick}
        to="/jest">
        {'jest'}
      </NavLink>
    </div>
  )
}

export default SliderContent

SliderContent.propTypes = {
  onItemClick: PropTypes.func.isRequired,
}
