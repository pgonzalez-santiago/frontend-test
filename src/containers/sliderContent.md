# Slider Content

----
## Props

* **onItemClick (required)**    Function: Callback function when a
item from the slider is clicked.

----
## Usage

```html
<SliderContent onItemClick={() => null}/>
```
