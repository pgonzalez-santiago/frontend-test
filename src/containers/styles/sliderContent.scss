@import '../../index.scss';

.slider-content {
  &.bm-item-list a {
    padding: 0.8em;
    display: block;
    outline: none;
    text-decoration: none;
    color: $primary-color;

    span {
      margin-left: 10px;
      font-weight: 700;
    }
  }
}
