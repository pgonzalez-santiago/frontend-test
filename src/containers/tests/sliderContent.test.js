import React from 'react'
import { SliderContent } from '../sliderContent'
import { shallow } from 'enzyme'

describe('SliderContent', () => {
  it('SliderContent Dom structure has not change when error', () => {
    const wrapper = shallow(
      <SliderContent
        error={true}
        fetching={false}
        getRepositories={() => null}
        onItemClick={() => null}
        pagination={{
          endCursor: '',
          hasNextPage: false,
        }}
        repositories={[]}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('SliderContent Dom structure has not change when fetching', () => {
    const wrapper = shallow(
      <SliderContent
        error={false}
        fetching={true}
        getRepositories={() => null}
        onItemClick={() => null}
        pagination={{
          endCursor: '',
          hasNextPage: false,
        }}
        repositories={[]}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('SliderContent Dom structure has not change when render repositories', () => {
    const wrapper = shallow(
      <SliderContent
        error={false}
        fetching={false}
        getRepositories={() => null}
        onItemClick={() => null}
        pagination={{
          endCursor: '',
          hasNextPage: false,
        }}
        repositories={[
          {
            name: 'react',
            stars: 116587,
            to: '/react',
            watchers: 6572
          },
          {
            name: 'react-native',
            stars: 71397,
            to: '/react-native',
            watchers: 3712
          },
          {
            name: 'create-react-app',
            stars: 59840,
            to: '/create-react-app',
            watchers: 1741
          },
        ]}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
