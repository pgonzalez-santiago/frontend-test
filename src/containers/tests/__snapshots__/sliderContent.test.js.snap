// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderContent SliderContent Dom structure has not change when error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SliderContent
    error={true}
    fetching={false}
    getRepositories={[Function]}
    onItemClick={[Function]}
    pagination={
      Object {
        "endCursor": "",
        "hasNextPage": false,
      }
    }
    repositories={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ForwardRef>
          <GithubLogo
            height={40}
            style={
              Object {
                "color": "#b8b7ad",
                "display": "block",
                "fill": "currentColor",
                "margin": "0 auto",
              }
            }
            width={40}
          />
        </ForwardRef>,
        <SliderList
          error={true}
          fetching={false}
          loadMore={[Function]}
          onItemClick={[Function]}
          pagination={
            Object {
              "endCursor": "",
              "hasNextPage": false,
            }
          }
          repositories={Array []}
        />,
      ],
      "className": "slider-content bm-item-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <GithubLogo
            height={40}
            style={
              Object {
                "color": "#b8b7ad",
                "display": "block",
                "fill": "currentColor",
                "margin": "0 auto",
              }
            }
            width={40}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "height": 40,
            "style": Object {
              "color": "#b8b7ad",
              "display": "block",
              "fill": "currentColor",
              "margin": "0 auto",
            },
            "width": 40,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": true,
            "rules": Array [
              "
  padding-top: 10px;
  padding-bottom: 20px;
  border-bottom: 1px solid ",
              "#b8b7ad",
              ";
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "error": true,
          "fetching": false,
          "loadMore": [Function],
          "onItemClick": [Function],
          "pagination": Object {
            "endCursor": "",
            "hasNextPage": false,
          },
          "repositories": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ForwardRef>
            <GithubLogo
              height={40}
              style={
                Object {
                  "color": "#b8b7ad",
                  "display": "block",
                  "fill": "currentColor",
                  "margin": "0 auto",
                }
              }
              width={40}
            />
          </ForwardRef>,
          <SliderList
            error={true}
            fetching={false}
            loadMore={[Function]}
            onItemClick={[Function]}
            pagination={
              Object {
                "endCursor": "",
                "hasNextPage": false,
              }
            }
            repositories={Array []}
          />,
        ],
        "className": "slider-content bm-item-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <GithubLogo
              height={40}
              style={
                Object {
                  "color": "#b8b7ad",
                  "display": "block",
                  "fill": "currentColor",
                  "margin": "0 auto",
                }
              }
              width={40}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "height": 40,
              "style": Object {
                "color": "#b8b7ad",
                "display": "block",
                "fill": "currentColor",
                "margin": "0 auto",
              },
              "width": 40,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-ifAKCX",
              "isStatic": true,
              "rules": Array [
                "
  padding-top: 10px;
  padding-bottom: 20px;
  border-bottom: 1px solid ",
                "#b8b7ad",
                ";
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-ifAKCX",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "error": true,
            "fetching": false,
            "loadMore": [Function],
            "onItemClick": [Function],
            "pagination": Object {
              "endCursor": "",
              "hasNextPage": false,
            },
            "repositories": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`SliderContent SliderContent Dom structure has not change when fetching 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SliderContent
    error={false}
    fetching={true}
    getRepositories={[Function]}
    onItemClick={[Function]}
    pagination={
      Object {
        "endCursor": "",
        "hasNextPage": false,
      }
    }
    repositories={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ForwardRef>
          <GithubLogo
            height={40}
            style={
              Object {
                "color": "#b8b7ad",
                "display": "block",
                "fill": "currentColor",
                "margin": "0 auto",
              }
            }
            width={40}
          />
        </ForwardRef>,
        <SliderList
          error={false}
          fetching={true}
          loadMore={[Function]}
          onItemClick={[Function]}
          pagination={
            Object {
              "endCursor": "",
              "hasNextPage": false,
            }
          }
          repositories={Array []}
        />,
      ],
      "className": "slider-content bm-item-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <GithubLogo
            height={40}
            style={
              Object {
                "color": "#b8b7ad",
                "display": "block",
                "fill": "currentColor",
                "margin": "0 auto",
              }
            }
            width={40}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "height": 40,
            "style": Object {
              "color": "#b8b7ad",
              "display": "block",
              "fill": "currentColor",
              "margin": "0 auto",
            },
            "width": 40,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": true,
            "rules": Array [
              "
  padding-top: 10px;
  padding-bottom: 20px;
  border-bottom: 1px solid ",
              "#b8b7ad",
              ";
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "error": false,
          "fetching": true,
          "loadMore": [Function],
          "onItemClick": [Function],
          "pagination": Object {
            "endCursor": "",
            "hasNextPage": false,
          },
          "repositories": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ForwardRef>
            <GithubLogo
              height={40}
              style={
                Object {
                  "color": "#b8b7ad",
                  "display": "block",
                  "fill": "currentColor",
                  "margin": "0 auto",
                }
              }
              width={40}
            />
          </ForwardRef>,
          <SliderList
            error={false}
            fetching={true}
            loadMore={[Function]}
            onItemClick={[Function]}
            pagination={
              Object {
                "endCursor": "",
                "hasNextPage": false,
              }
            }
            repositories={Array []}
          />,
        ],
        "className": "slider-content bm-item-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <GithubLogo
              height={40}
              style={
                Object {
                  "color": "#b8b7ad",
                  "display": "block",
                  "fill": "currentColor",
                  "margin": "0 auto",
                }
              }
              width={40}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "height": 40,
              "style": Object {
                "color": "#b8b7ad",
                "display": "block",
                "fill": "currentColor",
                "margin": "0 auto",
              },
              "width": 40,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-ifAKCX",
              "isStatic": true,
              "rules": Array [
                "
  padding-top: 10px;
  padding-bottom: 20px;
  border-bottom: 1px solid ",
                "#b8b7ad",
                ";
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-ifAKCX",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "error": false,
            "fetching": true,
            "loadMore": [Function],
            "onItemClick": [Function],
            "pagination": Object {
              "endCursor": "",
              "hasNextPage": false,
            },
            "repositories": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`SliderContent SliderContent Dom structure has not change when render repositories 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SliderContent
    error={false}
    fetching={false}
    getRepositories={[Function]}
    onItemClick={[Function]}
    pagination={
      Object {
        "endCursor": "",
        "hasNextPage": false,
      }
    }
    repositories={
      Array [
        Object {
          "name": "react",
          "stars": 116587,
          "to": "/react",
          "watchers": 6572,
        },
        Object {
          "name": "react-native",
          "stars": 71397,
          "to": "/react-native",
          "watchers": 3712,
        },
        Object {
          "name": "create-react-app",
          "stars": 59840,
          "to": "/create-react-app",
          "watchers": 1741,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ForwardRef>
          <GithubLogo
            height={40}
            style={
              Object {
                "color": "#b8b7ad",
                "display": "block",
                "fill": "currentColor",
                "margin": "0 auto",
              }
            }
            width={40}
          />
        </ForwardRef>,
        <SliderList
          error={false}
          fetching={false}
          loadMore={[Function]}
          onItemClick={[Function]}
          pagination={
            Object {
              "endCursor": "",
              "hasNextPage": false,
            }
          }
          repositories={
            Array [
              Object {
                "name": "react",
                "stars": 116587,
                "to": "/react",
                "watchers": 6572,
              },
              Object {
                "name": "react-native",
                "stars": 71397,
                "to": "/react-native",
                "watchers": 3712,
              },
              Object {
                "name": "create-react-app",
                "stars": 59840,
                "to": "/create-react-app",
                "watchers": 1741,
              },
            ]
          }
        />,
      ],
      "className": "slider-content bm-item-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <GithubLogo
            height={40}
            style={
              Object {
                "color": "#b8b7ad",
                "display": "block",
                "fill": "currentColor",
                "margin": "0 auto",
              }
            }
            width={40}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "height": 40,
            "style": Object {
              "color": "#b8b7ad",
              "display": "block",
              "fill": "currentColor",
              "margin": "0 auto",
            },
            "width": 40,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": true,
            "rules": Array [
              "
  padding-top: 10px;
  padding-bottom: 20px;
  border-bottom: 1px solid ",
              "#b8b7ad",
              ";
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "error": false,
          "fetching": false,
          "loadMore": [Function],
          "onItemClick": [Function],
          "pagination": Object {
            "endCursor": "",
            "hasNextPage": false,
          },
          "repositories": Array [
            Object {
              "name": "react",
              "stars": 116587,
              "to": "/react",
              "watchers": 6572,
            },
            Object {
              "name": "react-native",
              "stars": 71397,
              "to": "/react-native",
              "watchers": 3712,
            },
            Object {
              "name": "create-react-app",
              "stars": 59840,
              "to": "/create-react-app",
              "watchers": 1741,
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ForwardRef>
            <GithubLogo
              height={40}
              style={
                Object {
                  "color": "#b8b7ad",
                  "display": "block",
                  "fill": "currentColor",
                  "margin": "0 auto",
                }
              }
              width={40}
            />
          </ForwardRef>,
          <SliderList
            error={false}
            fetching={false}
            loadMore={[Function]}
            onItemClick={[Function]}
            pagination={
              Object {
                "endCursor": "",
                "hasNextPage": false,
              }
            }
            repositories={
              Array [
                Object {
                  "name": "react",
                  "stars": 116587,
                  "to": "/react",
                  "watchers": 6572,
                },
                Object {
                  "name": "react-native",
                  "stars": 71397,
                  "to": "/react-native",
                  "watchers": 3712,
                },
                Object {
                  "name": "create-react-app",
                  "stars": 59840,
                  "to": "/create-react-app",
                  "watchers": 1741,
                },
              ]
            }
          />,
        ],
        "className": "slider-content bm-item-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <GithubLogo
              height={40}
              style={
                Object {
                  "color": "#b8b7ad",
                  "display": "block",
                  "fill": "currentColor",
                  "margin": "0 auto",
                }
              }
              width={40}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "height": 40,
              "style": Object {
                "color": "#b8b7ad",
                "display": "block",
                "fill": "currentColor",
                "margin": "0 auto",
              },
              "width": 40,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-ifAKCX",
              "isStatic": true,
              "rules": Array [
                "
  padding-top: 10px;
  padding-bottom: 20px;
  border-bottom: 1px solid ",
                "#b8b7ad",
                ";
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-ifAKCX",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "error": false,
            "fetching": false,
            "loadMore": [Function],
            "onItemClick": [Function],
            "pagination": Object {
              "endCursor": "",
              "hasNextPage": false,
            },
            "repositories": Array [
              Object {
                "name": "react",
                "stars": 116587,
                "to": "/react",
                "watchers": 6572,
              },
              Object {
                "name": "react-native",
                "stars": 71397,
                "to": "/react-native",
                "watchers": 3712,
              },
              Object {
                "name": "create-react-app",
                "stars": 59840,
                "to": "/create-react-app",
                "watchers": 1741,
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
